lib/dispatcher.js

```diff
@@ -173,6 +176,24 @@ Dispatcher.prototype.dispatch = function(params, dispatchOptions) {
                 reject(error);
             });
 
+          } else if (error instanceof errors.Forbidden) {
+            // TODO: Remove this entire block once the API is consistent
+            // between 1.0 and 1.1.
+            try {
+              var errorStrs = error.value.errors.map(function(val) {
+                return val.message;
+              });
+              var errorMsg = error.message + ': ' + errorStrs.join('; ');
+              if (errorMsg.indexOf(PREMIUM_ONLY_STR) !== -1) {
```

Comment:
This works, but I wonder if we should even bother concatenating these strings with the `:` and `;`. `error.message` is something like `Forbidden` or `Payment Required`, so it will never contain `PREMIUM_ONLY_STR`, and for the rest you can just ask "do any of `errorStrs` have this substring?" Something like

```node
var matches = error.value.errors.some(function(error) {
  return error.message.indexOf(PREMIUM_ONLY_STR) !== -1
})
if (matches) { ... }
```

Reply:
Yeah I was just copying the technique from some clients for generating this string, and admit it was hastily done knowing it'll get stripped out. In the PHP one I went with a simpler method analogous to this. 🤷‍♂️